### PR TITLE
SAK-32478 ContentReview VeriCite to use encryption when uploading

### PR DESCRIPTION
**Pull request by peterdietz**
https://jira.sakaiproject.org/browse/SAK-32478

ContentReview implementation VeriCite to always encrypt uploads.

### CODE REVIEW

File: content-review/impl/vericite/src/main/java/org/sakaiproject/contentreview/vericite/ContentReviewServiceVeriCite.java [hunk 1]
```diff
@@ -910,6 +911,20 @@ private void uploadExternalContent(String urlString, byte[] data){
 			connection=(HttpURLConnection) url.openConnection();
 			connection.setDoOutput(true);
 			connection.setRequestMethod("PUT");
+
+			String headerString = ObjectUtils.toString(headers);
```
Comment (jonespm) [inline content-review/impl/vericite/src/main/java/org/sakaiproject/contentreview/vericite/ContentReviewServiceVeriCite.java]: It looks like this has been deprecated for  java.util.Objects.toString(Object)

File: content-review/impl/vericite/src/main/java/org/sakaiproject/contentreview/vericite/client/model/ExternalContentUploadInfo.java [hunk 2]
```diff
@@ -105,6 +108,24 @@ public void setFilePath(String filePath) {
...
+   /**
+   * A map of required headers
+   * @return headers
+  **/
+  @ApiModelProperty(example = "null", value = "A map of required headers")
+  public Object getHeaders() {
+    return headers;
+  }
+
+  public void setHeaders(Object headers) {
```
Comment (jonespm) [inline content-review/impl/vericite/src/main/java/org/sakaiproject/contentreview/vericite/client/model/ExternalContentUploadInfo.java]: Is this really going to be a variable Object? It kind of looks like from the code that it's expecting it to just be able to be easily convertible to a String?

File: content-review/impl/vericite/src/main/java/org/sakaiproject/contentreview/vericite/ContentReviewServiceVeriCite.java (hunk 1 above)
Comment (peterdietz) [inline content-review/impl/vericite/src/main/java/org/sakaiproject/contentreview/vericite/ContentReviewServiceVeriCite.java]: Thanks, didn't notice the deprecation warning in IDE. I think ObjectUtils came from commons-lang, which Sakai uses 2.6 (where it wasn't deprecated), as opposed to importing from commons-lang3, which is 3.5 where it is deprecated. I've replaced this with Objects.toString(headers, ""); I need to be ensure the value isn't null here.

File: content-review/impl/vericite/src/main/java/org/sakaiproject/contentreview/vericite/client/model/ExternalContentUploadInfo.java (hunk 2 above)
Comment (peterdietz) [inline content-review/impl/vericite/src/main/java/org/sakaiproject/contentreview/vericite/client/model/ExternalContentUploadInfo.java]: Object seems like a bad type for structured data. We are able to work with it just fine in string form, but you'll notice that we're doing some text manipulating. Removing curly brackets, splitting on a comma, splitting on equals sign, and setting a key and value. 

In the [api definition](https://github.com/vericite/LMS-API/blob/master/VeriCiteLmsApiV1.json#L104-L107), or a rest analog to a wsdl, the definition of the headers is that it is an object.

```
                "headers": {
                    "description": "A map of required headers",
                    "type": "object"
                },
```

**Pull request merged**